torch/distributed/_functional_collectives.py

```diff
@@ -109,7 +109,7 @@ class AsyncCollectiveTensor(torch.Tensor):
     Use it inside functional collective pytorch wrappers like the following:
     def functional_collective(self, group, tag):
         tag, rankset, group_size = _expand_group(group, tag)
-        tensor = torch._C._nn.{collective}(self, tag, rankset, group_size)
+        tensor = torch.ops.c10d_functional.{collective}(self, tag, rankset, group_size)
```

Comment:
a while ago you mentioned that it was quite slow to call ops.* and faster to call _C.* -- is this not a concern in the new approach?

Reply:
It might very much be, but `_C` is only available to ops defined in `native_functions.yaml` as it requires C++ codegen.

Reply:
Given collectives are quite rare in an optimized model, the usage of torch.ops should be ok.